File: vuebugtrackerproject.client/src/viewmodels/ProjectPreviewViewModel.ts
//View model for showing a preview of a project.
import AccountViewModel from "./AccountViewModel";

export default class ProjectPreviewViewModel{

	// Unique identifier for the project.
	ID: string = "";
	// The name of the project.
	Name: string = "";
	// The short description of the project.
	Summary: string = "";
	// The user that created the project.
	Owner: AccountViewModel;
	// The number of bugs in the project are open.
	OpenBugs: number = 0;
	// The total number of bugs in the project.
	TotalBugs: number = 0;
	// The date the project was last updated.
	DateModified: Date;
}

File: VueBugTrackerProject.Classes/ViewModels/ProjectPreviewViewModel.cs
namespace VueBugTrackerProject.Classes
{
	/// <summary>
	/// View model for a preview of a project.
	/// </summary>
	public class ProjectPreviewViewModel
	{
		/// <summary>
		/// Unique identifier for the project.
		/// </summary>
		public string ID { get; set; }
	
		/// <summary>
		/// The name of the project.
		/// </summary>
		public string Name { get; set; }
	
		/// <summary>
		/// The short description of the project.
		/// </summary>
		public string Summary { get; set; }
	
		/// <summary>
		/// The user that created the project.
		/// </summary>
		public AccountViewModel Owner { get; set; }
	
		/// <summary>
		/// The number of bugs in the project are open.
		/// </summary>
		public int OpenBugs { get; set; }
	
		/// <summary>
		/// The total number of bugs in the project.
		/// </summary>
		public int TotalBugs { get; set; }
	
		/// <summary>
		/// The date the project was last updated.
		/// </summary>
		public DateTime DateModified { get; set; }

		/// <summary>
		/// Creates a view model summary of a project.
		/// </summary>
		/// <param name="project">The project the view model is being made for.</param>
        public ProjectPreviewViewModel(Project project)
        {
            ID = project.ID;
			Name = project.Name;
			Summary = project.Summary;
			Owner = new AccountViewModel(project.Owner);
			OpenBugs = project.Bugs.Count(b => b.Status == Status.Open);
			TotalBugs = project.Bugs.Count();
			DateModified = project.DateModified;
        }

    }
}
